subprojects/core/src/main/java/org/gradle/internal/service/scopes/ExecutionGradleServices.java

```diff
@@ -122,53 +122,42 @@ OutputChangeListener createOutputChangeListener(ListenerManager listenerManager)
-        TimeoutHandler timeoutHandler
+        OverlappingOutputDetector overlappingOutputDetector,
+        TimeoutHandler timeoutHandler,
+        ValueSnapshotter valueSnapshotter
     ) {
-        return new DefaultWorkExecutor<AfterPreviousExecutionContext, CachingResult>(
-            new CaptureStateBeforeExecutionStep(classLoaderHierarchyHasher, valueSnapshotter,
-                new ResolveCachingStateStep(buildCacheController, buildScanPlugin.isBuildScanPluginApplied(),
-                    new MarkSnapshottingInputsFinishedStep<UpToDateResult>(
-                        new ResolveChangesStep<UpToDateResult>(changeDetector,
-                            new SkipUpToDateStep<IncrementalChangesContext>(
-                                new RecordOutputsStep<IncrementalChangesContext>(outputFilesRepository,
-                                    new StoreSnapshotsStep<IncrementalChangesContext>(
-                                        new BroadcastChangingOutputsStep<IncrementalChangesContext, CurrentSnapshotResult>(outputChangeListener,
-                                            new CacheStep(buildCacheController, buildCacheCommandFactory,
-                                                new SnapshotOutputsStep<IncrementalChangesContext>(buildInvocationScopeId.getId(),
-                                                    new CreateOutputsStep<IncrementalChangesContext, Result>(
-                                                        new CatchExceptionStep<IncrementalChangesContext>(
-                                                            new TimeoutStep<IncrementalChangesContext>(timeoutHandler,
-                                                                new CancelExecutionStep<IncrementalChangesContext>(cancellationToken,
-                                                                    new ResolveInputChangesStep<IncrementalChangesContext>(
-                                                                        new CleanupOutputsStep<InputChangesContext, Result>(
-                                                                            new ExecuteStep<InputChangesContext>()
-                                                                        )
-                                                                    )
-                                                                )
-                                                            )
-                                                        )
-                                                    )
-                                                )
-                                            )
-                                        )
-                                    )
-                                )
-                            )
-                        )
-                    )
-                )
-            )
-        );
+        return new DefaultWorkExecutor<>(
+            new LoadPreviousExecutionStateStep<>(
```

Comment:
Every time someone indents the file, all this will be lost again. Should we disable formatting in that block via 
```
// @formatter:off
...
// @formatter:on
```

Isn't there a better way to add all this decoration without nesting the calls so much?

Reply:
Good point, added.